harness/determined/profiler.py

```diff
@@ -81,22 +101,33 @@ def __init__(
         local_rank: int,
         start_on_batch: int,
         end_after_batch: Optional[int] = None,
+        send_batch_fn: SendBatchFnType = api.post_trial_profiler_metrics_batches,
+        check_data_exists_fn: CheckDataExistsFnType = profiling_metrics_exist,
     ):
         self.current_batch_idx = 0
-        self.agent_id = agent_id
         self.trial_id = trial_id
+        self.agent_id = agent_id
         self.master_url = master_url
+        self.profiling_is_enabled_in_experiment_config = profiling_is_enabled
+        self.global_rank = global_rank
+        self.local_rank = local_rank
         self.start_on_batch = start_on_batch
         self.end_after_batch = end_after_batch
-        self.local_rank = local_rank
-        self.global_rank = global_rank
-
-        self.profiling_is_enabled_in_experiment_config = profiling_is_enabled
+        self.send_batch_fn = send_batch_fn
+        self.check_data_already_exists_fn = check_data_exists_fn
```

Comment:
Is this going to be a convenience for testing?

Reply:
Yeah, writing unit tests is much easier if we can replace the functions that communicate with the master when creating the `ProfilingAgent`